docs/getting-started/how-to/add-github-codespaces-to-your-project/index.md

```diff
@@ -0,0 +1,126 @@
+This page describes how to add GitHub Codespaces to your project.
+
+## Does your project already have GitHub Codespaces?
+
+Newer projects already have GitHub Codespaces.
+However, older projects don't.
+To find out whether your project is newer or older:
+
+* Click on your project in the [opensafely](https://github.com/opensafely) organisation on GitHub.
+* Can you see a `.devcontainer` folder?
+  If not, then your project is **older**.
+* Click on the `.devcontainer` folder to open it.
+* Click on the `devcontainer.json` file to open it.
+* Can you see the following line towards the top of the file?
+  ```json
+  "image": "ghcr.io/opensafely-core/research-template:v0",
+  ```
+  If not, then your project is **older**.
+* If you've got this far, then your project is **newer**.
+
+## Adding GitHub Codespaces to your project
+
+We will use Git to add GitHub Codespaces to your project.
+There are many ways to use Git.
+For example, we could use Git with the `git` command-line tool, with GitHub Desktop, or with VS Code.
+Here, we will use Git with `git`, because `git` commands are succinct.
+The documentation for
+[GitHub Desktop](https://docs.github.com/en/desktop) and
+[VS Code](https://code.visualstudio.com/docs/sourcecontrol/overview)
+describes how to translate `git` commands into equivalent user-interface commands.
+
+Here, you will run `git` commands on your computer.
+
+### Identify your project's working branch
+
+The *working branch* contains work that is "done".
+It is often called `main` and it is often associated with a workspace on OS Jobs.
+
+We will add GitHub Codespaces to the working branch.
+
+### Create a new branch
+
+Switch to the working branch:
+
+```sh
+git switch <working branch>
+```
+
+Create a new branch called `github-codespaces`:
+
+```sh
+git switch --create=github-codespaces
+```
+
+### Copy files
+
+* Visit the [opensafely/research-template](https://github.com/opensafely/research-template) project on GitHub.
+* Click on the `.devcontainer` folder to open it.
+* Click on the first file to open it.
+* Click on "Download raw file" to download the first file to your computer.
```

Comment:
This could be improved with an image of the logo - I didn't know which button to click! 
![image](https://github.com/opensafely/documentation/assets/25795240/9958e1a0-ce62-416c-89f8-dd240126843b)

Reply:
Added! Thanks for the suggestion.